src/Tes.RunnerCLI/Commands/CommandFactory.cs

```diff
@@ -7,67 +10,81 @@ internal class CommandFactory
     {
         const string DefaultTaskDefinitionFile = "TesTask.json";
 
-        private static readonly Uri DefaultDockerUri = new Uri("unix:///var/run/docker.sock");
+        private static readonly Uri DefaultDockerUri = new("unix:///var/run/docker.sock");
 
         internal const string UploadCommandName = "upload";
         internal const string DownloadCommandName = "download";
         internal const string DockerUriOption = "docker-url";
+        internal const string SkipMissingSources = "skipMissingSources";
 
-        internal static RootCommand CreateExecutorCommand()
+        private static readonly IReadOnlyCollection<Option> GlobalOptions = new List<Option>()
         {
-            var dockerUriOption = CreateOption<Uri>(CommandFactory.DockerUriOption, "local docker engine endpoint", "-u", defaultValue: DefaultDockerUri);
+            CreateOption<FileInfo>(BlobPipelineOptionsConverter.FileOption, "The file with the task definition",  "-f", required: true, defaultValue: GetDefaultTaskDefinitionFile()),
+            CreateOption<int>(BlobPipelineOptionsConverter.BlockSizeOption, "Blob block size in bytes", "-b", defaultValue: BlobSizeUtils.DefaultBlockSizeBytes),
+            CreateOption<int>(BlobPipelineOptionsConverter.WritersOption, "Number of concurrent writers", "-w", defaultValue: BlobPipelineOptions.DefaultNumberOfWriters),
+            CreateOption<int>(BlobPipelineOptionsConverter.ReadersOption, "Number of concurrent readers", "-r", defaultValue: BlobPipelineOptions.DefaultNumberOfReaders),
+            CreateOption<int>(BlobPipelineOptionsConverter.BufferCapacityOption, "Pipeline buffer capacity", "-c", defaultValue: BlobPipelineOptions.DefaultReadWriteBuffersCapacity),
+            CreateOption<string>(BlobPipelineOptionsConverter.ApiVersionOption, "Azure Storage API version", "-v", defaultValue: BlobPipelineOptions.DefaultApiVersion)
+        }.AsReadOnly();
+
+        private static readonly Option SkipMissingSourcesOption = CreateOption<bool>(CommandFactory.SkipMissingSources, "Skip missing sources for file transfers", "-s", defaultValue: false); // TODO: remove this option when we incorporate the batch node script logic into this executable
```

Comment:
Is adding this option the only reason that lead to the refactoring of the CLI?  That seems out of scope of the issue. Specially if the option seems to be temporary.

Reply:
Yes, but as discussed we have a different route to solve this.